server/src/main/java/org/opensearch/rest/action/admin/cluster/RestNodesStatsAction.java

```diff
@@ -175,6 +176,25 @@ public RestChannelConsumer prepareRequest(final RestRequest request, final NodeC
 
                     nodesStatsRequest.indices(flags);
                 }
+            } else if (metrics.contains("caches")) {
+                // Extract the list of caches we want to get stats for from the submetrics (which we get from index_metric)
+                Set<String> cacheMetrics = Strings.tokenizeByCommaToSet(request.param("index_metric", "_all"));
```

Comment:
Seems like running the above returns: 

`{"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"request [/_nodes/stats/request_cache] contains unrecognized metric: [request_cache] -> did you mean any of [file_cache, script_cache]?"}],"type":"illegal_argument_exception","reason":"request [/_nodes/stats/request_cache] contains unrecognized metric: [request_cache] -> did you mean any of [file_cache, script_cache]?"},"status":400}`

which I believe is correct. This comes from line 151 which checks for invalid metrics before submetrics are checked (so the change here doesn't affect it).

Reply:
Ok, after looking into how `index_metric` gets assigned, I'm pretty sure it's fine to use it, and we actually can't/shouldn't create our own `cache_metric`. 

Here `index_metric` ends up in the params eventually, but it's just a label for a certain wildcard name in a URI path. The allowed URIs for stats actions are defined in routes() of RestNodesStatsAction: 

```public List<Route> routes() {
        return unmodifiableList(
            asList(
                new Route(GET, "/_nodes/stats"),
                new Route(GET, "/_nodes/{nodeId}/stats"),
                new Route(GET, "/_nodes/stats/{metric}"),
                new Route(GET, "/_nodes/{nodeId}/stats/{metric}"),
                new Route(GET, "/_nodes/stats/{metric}/{index_metric}"),
                new Route(GET, "/_nodes/{nodeId}/stats/{metric}/{index_metric}")
            )
        );
    }
```

The last two routes are the relevant ones for us. If we were to make a new `cache_metric` instead of `index_metric` we would theoretically add `new Route(GET, "/_nodes/{nodeId}/stats/{metric}/{cache_metric}")`. But, given two routes which only differ by the name of their wildcard, there's no way for the handlers to determine which wildcard to use. Indeed if we do this the cluster crashes on start with the error `java.lang.IllegalArgumentException: Trying to use conflicting wildcard names for same path: index_metric and cache_metric`

Ultimately since `index_metric` is just a label, it should be fine to use regardless. It might be nice to rename it to something more generic like `submetric`, but I don't think we should because the value "index_metric" is hardcoded in various places and I'm not sure we would catch them all.